### DISCUSSION TITLE
Maximum mute duration (other than indefinite)

### DISCUSSION THREAD

**Discussion by InsertUser** [mastodon/mastodon#31794] [Q&A]
At the moment the two interfaces I use for the Mastodon both have the maximum duration of a (finite) mute set to "7 days". To mute longer you have to choose "indefinite" and remember to unmute them manually. 

According to the accounts API page for [mute](https://docs.joinmastodon.org/methods/accounts/#mute) this is actually sent to the server  as a duration:

> Number. How long the mute should last, in seconds. Defaults to 0 (indefinite).

Is there a limit on how large this number can be?

Are there any known issues with limited time mute durations above 7 days?

Is there anything server side that would substitute a duration that's "too long" either for a shorter duration or replace it with an indefinite mute automatically?

**Comment by mjankowski** [comment]
I'm mostly sure this is accurate...

When that value comes in, it does a few relevant things:

- On the save mute itself (connected to account), sets an expires_at value
- Enqueueus a job to delete the mute after the duration seconds value

I don't think there is numerical validation or limitation on either of these. In some local API testing I was able to create a mute which expires in the year 33713 via api duration param.

Presumably with a large enough number you might hit some sort of number or date overflow/validity issue either in the db or ruby itself or the job queue, etc ... but I think for non-absurd values there's not a limit.
